tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py

```diff
@@ -58,22 +62,30 @@ def configure(self):
         self.data_dir = self.get_value_from_config('data_dir')
         self.in_suffix = self.get_value_from_config('input_suffix')
         self.out_suffix = self.get_value_from_config('target_suffix')
+        self.recursive = self.get_value_from_config('recursive')
         self.annotation_loader = LOADERS_MAPPING.get(self.get_value_from_config('annotation_loader'))
         if not self.annotation_loader:
             raise ConfigError('provided not existing loader')
 
     def convert(self, check_content=False, progress_callback=None, progress_interval=100, **kwargs):
         content_errors = [] if check_content else None
         file_list_in = []
-        for file_in_dir in self.data_dir.iterdir():
+        if self.recursive:
+            data_dir_files = [file for file in self.data_dir.glob('**/*') if file.is_file()]
+        else:
+            data_dir_files = self.data_dir.iterdir()
+        for file_in_dir in data_dir_files:
             if self.in_suffix in file_in_dir.parts[-1]:
                 file_list_in.append(file_in_dir)
 
         annotation = []
         num_iterations = len(file_list_in)
         for in_id, in_file in enumerate(file_list_in):
-            in_file_name = in_file.parts[-1]
-            gt_file_name = self.out_suffix.join(in_file_name.split(self.in_suffix))
+            in_file_name = in_file.relative_to(self.data_dir) if self.recursive else in_file.parts[-1]
+            if self.recursive:
+                gt_file_name = Path(self.out_suffix.join(str(in_file_name).split(self.in_suffix)))
```

Comment:
why do you need to wrap into Path?
it is unsafe store path in pickle (Path class is not OS agnostic)

Reply:
I believe, after these suggestion  https://github.com/openvinotoolkit/open_model_zoo/pull/2199/files#r594196221
behaviour for recursive and non-recursive will be the same